### PR TITLE
Support scrolling the main content by fixing top and bottom nav

### PR DESCRIPTION
**Pull request by bqx**
Also:
* Fix up the text in the top nav so it now renders inside the buttons
* Minor tweaks to sizing of footer elements
* Force Talents to render at bottom of main, scrollable content (e.g. when the Inventory is empty)

### CODE REVIEW

**Review by rissole — CHANGES_REQUESTED** [review]
image

modals broke

File: src/App.scss [hunk 1]
```diff
@@ -30,18 +18,6 @@ $navButtonHeight: 72px;
   align-self: flex-end;
 }
 
-.main {
```
Comment (AndaxH) [inline src/App.scss:33]: Question: Were these styles meant to be removed for `.nav` and `.main` ?

File: src/components/LayoutFooter/ClassText/index.tsx [hunk 2]
```diff
@@ -23,7 +19,12 @@ const CLASS_TO_COLOR: { [key in HeroClass]: string } = {
...
 const ClassText = ({ heroClass }: Props) => {
   const text = heroClass.substring(0, 1).toUpperCase().concat(heroClass.substring(1));
-  return <ClassColorText color={CLASS_TO_COLOR[heroClass]}>{text}</ClassColorText>;
+  return <TextContainer color={CLASS_TO_COLOR[heroClass]}>{text}</TextContainer>;
 };
 
 export default ClassText;
+
+const TextContainer = styled.span<{ color: string }>`
+  color: ${(props) => props.color || "#000000"};
```
Comment (AndaxH) [inline src/components/LayoutFooter/ClassText/index.tsx]: Minor Suggestion: Would be nice for #000000 to be a constant if it's a default

File: src/components/LayoutHeader/TopNavItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  );
+};
+
+export default TopNavItem;
+
+const TopNavContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 2px;
```
Comment (AndaxH) [inline src/components/LayoutHeader/TopNavItem/index.tsx:33]: Question: Is gap applied for `display: flex` elements? I thought it was `display: grid` specific

File: src/components/ListInventory/index.tsx [hunk 4]
```diff
@@ -43,6 +44,10 @@ const InventoryList = () => {
     setSelectedCardForModal(null);
   }, []);
 
+  if (inventoryCardsData.length < 1) {
+    return <NoItemsMessage>アイテムはありません。</NoItemsMessage>;
```
Comment (AndaxH) [inline src/components/ListInventory/index.tsx]: Question: What does this mean?

File: src/components/ScreenMain/index.tsx [hunk 5]
```diff
@@ -60,83 +29,37 @@ const MainScreen = () => {
...
-    filter: grayscale(100%);
-  }
+const MainContainer = styled.div`
+  width: 100%;
+  display: flex;
+  flex-flow: column nowrap;
+  flex: 1 1 auto;
+  margin-top: ${LAYOUT.navHeight}px;
+  margin-bottom: ${LAYOUT.footerHeight}px;
+  overflow-y: scroll;
```
Comment (AndaxH) [inline src/components/ScreenMain/index.tsx]: Suggestion: Change this to `overflow-y: auto` which will only show scroll bars when overflowing as opposed to always showing scroll bars regardless of height

File: src/util/layout.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const LAYOUT = {
+  // Height of header in px
+  navHeight: 69,
```
Comment (AndaxH) [inline src/util/layout.ts:3]: nice

**Review by AndaxH — COMMENTED** [review]
Love your work Ben - left some suggestions and questions but mostly minor stuff

File: src/App.scss (hunk 1 above)
Comment (bqx) [reply src/App.scss:33]: Yeah I moved them into a styled component instead

File: src/components/ListInventory/index.tsx (hunk 4 above)
Comment (bqx) [reply src/components/ListInventory/index.tsx]: @rissole Is this the right particle?

File: src/util/layout.ts (hunk 6 above)
Comment (bqx) [reply src/util/layout.ts:3]: nice

File: src/components/LayoutHeader/TopNavItem/index.tsx (hunk 3 above)
Comment (bqx) [reply src/components/LayoutHeader/TopNavItem/index.tsx:33]: yar it is, see: https://css-tricks.com/almanac/properties/g/gap/

File: src/components/ListInventory/index.tsx (hunk 4 above)
Comment (rissole) [reply src/components/ListInventory/index.tsx]: が i think puts more emphasis on the items but it's all good

edit: i did a bit of googling and が seems to be used in some twitter messages so 🤷

File: src/components/LayoutHeader/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        <TopNavItem className="talents" iconPath={talentsIcon} onClick={toggleTalentModal} label="Talents" />
+        <TopNavItem className="items" iconPath={shopIcon} onClick={() => console.log("Items Modal")} label="Items" />
+        <TopNavItem
+          active={activeList === "reference"}
+          className="more"
+          iconPath={referenceIcon}
+          onClick={toggleListBetweenPowersAndReference}
+          label="Info"
+        />
+      </NavContainer>
```
Comment (rissole) [inline src/components/LayoutHeader/index.tsx:54]: the point of the text is to disambiguate the functionality, at the moment i dont feel like the text achieves that! I understand going for brevity but we have plenty of room and i think it looks ok.

image

* Spells -> Train Spells or just "Train"
* Talents -> All Talents
* Items -> Item shop
* Info -> Reference

File: src/components/EmptyCharacterSheetSlot/index.tsx [hunk 8]
```diff
@@ -96,8 +96,8 @@ const EmptySlotTypeIcons = ({ slotTypes }: { slotTypes: CardSlotMetadata["slotTy
                 height: "42px",
                 width: "42px",
                 opacity: "70%",
-                borderRadius: "8px",
-                border: "1px solid black",
+                borderRadius: "20px",
```
Comment (rissole) [inline src/components/EmptyCharacterSheetSlot/index.tsx]: 2px i think.

**Review by rissole — CHANGES_REQUESTED** [review]
loving the easter egg nihongo

File: src/components/EmptyCharacterSheetSlot/index.tsx (hunk 8 above)
Comment (bqx) [reply src/components/EmptyCharacterSheetSlot/index.tsx]: Have made this consistent with the EditableStat radius + box shadow styles

File: src/components/LayoutHeader/index.tsx (hunk 7 above)
Comment (bqx) [reply src/components/LayoutHeader/index.tsx:54]: Yeah I like that, will do in follow up PR

**Review by rissole — APPROVED** [review]

**Pull request merged**